src/processor/operator/persistent/reader/csv/driver.cpp

```diff
@@ -85,51 +85,55 @@ BaseCSVReader* SerialParsingDriver::getReader() {
     return reader;
 }
 
-bool SniffCSVNameAndTypeDriver::done(uint64_t) {
-    return true;
-}
-
-void SniffCSVNameAndTypeDriver::addValue(uint64_t, common::column_id_t, std::string_view value) {
-    std::string columnName(value);
-    LogicalType columnType(LogicalTypeID::STRING);
-    auto it = value.rfind(':');
-    if (it != std::string_view::npos) {
-        try {
-            columnType =
-                context->getCatalog()->getType(context->getTx(), std::string(value.substr(it + 1)));
-            columnName = std::string(value.substr(0, it));
-        } catch (const Exception&) { // NOLINT(bugprone-empty-catch):
-                                     // This is how we check for a suitable
-                                     // datatype name.
-            // Didn't parse, just use the whole name.
-        }
+bool SniffCSVNameAndTypeDriver::done(uint64_t rowNum) const {
+    if (csvOptions.sampleSize == -1) {
+        return false;
     }
-    columns.emplace_back(columnName, std::move(columnType));
-}
-
-bool SniffCSVNameAndTypeDriver::addRow(uint64_t, common::column_id_t) {
-    return true;
-}
-
-bool SniffCSVColumnCountDriver::done(uint64_t) const {
-    return !emptyRow;
+    return (csvOptions.hasHeader ? 1 : 0) + csvOptions.sampleSize <= (int64_t)rowNum;
 }
 
-void SniffCSVColumnCountDriver::addValue(uint64_t, common::column_id_t columnIdx,
+void SniffCSVNameAndTypeDriver::addValue(uint64_t rowNum, common::column_id_t columnIdx,
```

Comment:
So we don't seem to be able to sniff delimiter, line breaker, etc.

Reply:
If you mean CSVs with delimiters & line breakers inside elements, I think we'll be able to handle that. That part of the code is separate from the driver.